### PR TITLE
feat: OPA build from source

### PR DESCRIPTION
**Pull request by fhennig**
# Description

We tested that the smoke test runs :+1: 

## Definition of Done Checklist

- Not all of these items are applicable to all PRs, the author should update this template to only leave the boxes in that are relevant
- Please make sure all these things are done and tick the boxes
 
```[tasklist]
- [ ] Changes are OpenShift compatible
- [ ] All added packages (via microdnf or otherwise) have a comment on why they are added
- [ ] Things not downloaded from Red Hat repositories should be mirrored in the Stackable repository and downloaded from there
- [ ] All packages should have (if available) signatures/hashes verified
- [ ] Does your change affect an SBOM? Make sure to update all SBOMs
- [ ] Add an entry to the CHANGELOG.md file
- [ ] Integration tests ran successfully
```

<details>
<summary>TIP: Running integration tests with a new product image</summary>

The image can be built and uploaded to the kind cluster with the following commands:

```shell
bake --product <product> --image-version <stackable-image-version>
kind load docker-image <image-tagged-with-the-major-version> --name=<name-of-your-test-cluster>
```

See the output of `bake` to retrieve the image tag for `<image-tagged-with-the-major-version>`.
</details>

### CODE REVIEW

File: opa/Dockerfile [hunk 1]
```diff
@@ -58,13 +58,32 @@ WORKDIR /daemontools/admin/daemontools-${DAEMONTOOLS_VERSION}
...
 ARG PRODUCT
 ARG RELEASE
 ARG TARGETARCH
 ARG TARGETOS
 
+ENV GOARCH=$TARGETARCH
+ENV GOOS=$TARGETOS
+
+RUN microdnf update && \
+    microdnf install git go gzip tar
```
Comment (Maleware) [inline opa/Dockerfile:72]:
```suggestion
    microdnf install --assumeyes \
        git \
        go \
        gzip \
        tar && \
        microdnf clean all
```

Comment (Maleware) [reply opa/Dockerfile:72]: We use as convention one line per installed tool. I would personally appreciate a comment why / for what you install those tools.

File: opa/Dockerfile [hunk 2]
```diff
@@ -58,13 +58,32 @@ WORKDIR /daemontools/admin/daemontools-${DAEMONTOOLS_VERSION}
 
 RUN package/install
 
-FROM stackable/image/vector
+FROM stackable/image/vector as opa-builder
```
Comment (Maleware) [inline opa/Dockerfile:61]: I think stackable-base is sufficient as a builder image at this point.

Comment (Maleware) [reply opa/Dockerfile:61]: You would need to add 
```
"stackable-base": "1.0.0",
```
to `conf.py` tested it locally and it was sufficient. you can also add stackable-base to opa-bundle-builder afaik, but @siegfriedweber might have a veto for that.

**Review by Maleware — CHANGES_REQUESTED** [review]
Thanks for your work, just a few nitty things.

File: opa/Dockerfile (hunk 2 above)
Comment (lfrancke) [reply opa/Dockerfile:61]: And could you add a hadolint ignore to this line? Look at Kafka for an example.

File: opa/Dockerfile (hunk 1 above)
Comment (lfrancke) [reply opa/Dockerfile:72]: Yes, Maxi is right.
Please split it across multiple lines.
`--assumeyes` is not needed. That is now baked into the base image and also yes: Please add a comment for why everything is needed. Thanks!

File: opa/Dockerfile [hunk 3]
```diff
@@ -58,13 +58,32 @@ WORKDIR /daemontools/admin/daemontools-${DAEMONTOOLS_VERSION}
...
+
+RUN curl --fail -L "https://repo.stackable.tech/repository/packages/opa/opa_${PRODUCT}.tar.gz" -o opa.tar.gz && \
+    tar -zxvf opa.tar.gz && \
+    mv opa-${PRODUCT} opa
+
+WORKDIR /opa
+
+RUN go build -o opa -buildmode=exe
+
+FROM stackable/image/vector
```
Comment (lfrancke) [inline opa/Dockerfile:82]: See comment above about the hadolint ignore

**Review by lfrancke — CHANGES_REQUESTED** [review]

File: opa/Dockerfile (hunk 1 above)
Comment (fhennig) [reply opa/Dockerfile:72]: Will do. Is `microdnf clean all` really needed? This is just a builder image, so it gets thrown away anyways

**Review by Maleware — APPROVED** [review]
Another Job well done :) Thanks!

**Pull request merged**